### PR TITLE
[New] Add openui5-sap.ui.core@1.50.0 w/ npm auto-update

### LINKED ISSUE #9244: [Request] Add OpenUI5
**Library name:** OpenUI5
**Git repository url:** https://github.com/SAP/openui5
**npm package url(optional):**  none
**License(s):** https://github.com/SAP/openui5/blob/master/LICENSE.txt
**Official homepage:** http://openui5.org/
**Wanna say something? Leave message here:**
would be nice to have this framework available

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38269129-request-add-openui5?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sufuf3**
### Memo
In this page http://openui5.org/download.html we know that we have to add `sap-ui-core.js`
There is no `sap-ui-core.js` in main repo.
Via discuss with the author, SAP/openui5#1237
We know that we need to add this file from https://github.com/openui5/packaged-sap.ui.core
So I add the file via this repo.

the description part, I combine https://github.com/SAP/openui5/blob/master/package.json#L4 and https://github.com/openui5/packaged-sap.ui.core/blob/rel-1.44/bower.json#L3

---

close #9244

Pull request for issue: #9244 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/SAP/openui5
 * Official website (optional, not the repository): http://openui5.org/
 * NPM package url (optional): none
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 295
   - Count of stars: 1387
   - Count of forks: 681

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by matz3** [comment]
Maintainer of OpenUI5 here 👋  

I'm not sure if I get this right, but this means only the `sap-ui-core.js` file will be included for CDNJS, right?

I don't think that this will be sufficient to use OpenUI5 from there. The `sap-ui-core.js` is the bootstrap file which will start-up the framework and provide everything required to e.g. load themes or further libraries. It is a bundle of all files required for the [`sap/ui/core/Core`](https://github.com/openui5/packaged-sap.ui.core/blob/rel-1.50/resources/sap/ui/core/Core.js) class.
The `sap.ui.core` library will always be loaded (see files from https://github.com/openui5/packaged-sap.ui.core/tree/rel-1.50/resources/sap/ui/core).
Other libraries, e.g. [`sap.m`](https://github.com/openui5/packaged-sap.m) (which provides most of the commonly used UI elements) might be loaded as well, depending on what you need for your application.

We already create bundles for each of those librares, called `library-preload.js`:
[sap.ui.core](https://github.com/openui5/packaged-sap.ui.core/blob/rel-1.50/resources/sap/ui/core/library-preload.js) / [sap.m](https://github.com/openui5/packaged-sap.m/blob/rel-1.50/resources/sap/m/library-preload.js)
Including those bundles should be sufficient, so not all individual source files need to be added.

Here is a simple Hello World App:
https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/helloworld/index.html

In case of any questions, feel free to reach out to me 👍

**Comment by sufuf3** [comment]
@matz3 Thank you for your detail comment.
It's a little busy recently.
I'll fix this PR in this week.

Thank you.

**Comment by sufuf3** [comment]
@matz3 Thank you for your update.
Sorry for pending this PR. Would you please let me know which auto-update do you like, Git or NPM? 
If you choose NPM, I would waiting for lib on npm. 
Or, maybe I can do this first. After the lib publish on npm, I will help update the auto-update.

Thank you.

**Comment by matz3** [comment]
@sufuf3 I think npm would make more sense.

**Comment by sufuf3** [comment]
Got it.
Thank you for your response.

File: ajax/libs/openui5-sap.ui.core/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/SAP/openui5.git"
+  },
+  "npmName": "@openui5/sap.ui.core",
+  "npmFileMap": [
+    {
+      "basePath": "sap.ui.core/src",
+      "files": [
+        "sap-ui-core.js"
```
Comment (matz3) [inline ajax/libs/openui5-sap.ui.core/package.json]: As mentioned in https://github.com/cdnjs/cdnjs/pull/11789#issuecomment-328535263 this is not sufficient.
Also, the npm packages currently don't contain bundles at all.

Comment (MattIPv4) [reply ajax/libs/openui5-sap.ui.core/package.json]: Hey @matz3 - where can we pull the bundles in from, do you have tagged release on GitHub that contain the bundled versions?

Adittionally, could you provide a full list of what files we should be including here?

Comment (matz3) [reply ajax/libs/openui5-sap.ui.core/package.json]: No, we don't have the bundles within npm or Github right now. We consider adding them to the npm packages or to provide additional packages in future.

The only way would be to build them manually or fetch them from our CDN (https://openui5.hana.ondemand.com).

To be on the safe side you would need to include all files within the "src" folder.
In addition to that, just adding the sap.ui.core library won't be sufficient, as more libraries are required to use OpenUI5 (see https://www.npmjs.com/org/openui5).

Also, the URL paths will be different than expected by default so users need to configure paths to the individual libraries in order to properly load them.

While I really like the idea to have a central CDN to host all kind of JavaScript frameworks and libraries I don't quite see the benefit of adding OpenUI5 as there is already a CDN and the steps to add it are not as straightforward as for other, rather simple libraries just consisting of one JS file.

Comment (MattIPv4) [reply ajax/libs/openui5-sap.ui.core/package.json]: Makes sense. I think I agree that adding it here is more work than the benefit it gives 👍

**Comment by MattIPv4** [comment]
Per the discussion above, adding this library is far more complicated than originally thought whilst offereing little benefit. As such, closing.

**Pull request closed without merging**